MIPs/new-protocol_metadata.md

```diff
@@ -0,0 +1,159 @@
+  - Or an Atomic type defined above
+
+### Schema
+
+A Mercury Schema is a set of properties (with types) that are either present in a given profile or not. If required properties are missing, then that profile does not match/represent this schema. There can be optional properties in a schema that might be present or not. If a property exists with the same key, but different type, then it is not a match. For optional properties, these non-matches must be treated as missing, for required properties, such a non-match means failure of matching the whole schema.
+
+By representing the schema as a flat list of properties developers can easily combine MIP based shared schemas with their own dApp defined schemas into a single server request for metadata or alternately a fine grained composition tailored to the specific needs of a view in a UI.
+
+The schema should not be transferred together with the profile, it is an independent definition. Applications and use-cases should define schemas and match them to profiles. So profiles are not strongly typed, and just looking at a profile it does not need to describe what schemas it implements.
+
+## Specification
+
+Cap'n Proto partial definition -
+
+```capnproto
+struct MetaProperty {
+    # typed key for key-value dictionary/map container
+
+    key @0 :Text;
+    # The unique name of the property in the mapping. This name can have multiple forms:
+    #   - MIP defined, ex. 'MIP3.Persona.Username', 'MIP3.Home.Addresses',
+    #        'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #        'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+
+    atomic @1 :Text;
+    # semantic is a string that represents an optional codec-like type definition
+    # semantic types have similar forms to Keys:
+    #   - MIP defined, ex. 'MIP3:Person:Name', 'MIP3:Address', 'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #       'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+    #   - An Atomic type defined above
+
+    semantic @1 :Text;
+    # semantic is a string that represents an optional codec-like type definition
+    # semantic types have similar forms to Keys:
+    #   - MIP defined, ex. 'MIP3:Person:Name', 'MIP3:Address', 'MIP3:Home:MultiAddress'
+    #   - dApp defined, ex. 'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Actor.Username',
+    #       'APP.139d2275-1d3b-4ba7-9d28-9a1231e7f49d.Sparks'
+    #   - An Atomic type defined above
+}
+
+struct MetaSchema {
+    # A schema is defined by a list of MetaProperty definitions
+
+    types @0 :List(MetaProperty);
+}
+
+struct MetaData {
+    # MetaProperty/value dictionary/map container, stores all available shapes
+
+    entries @0 :List(Entry);
+    struct Entry
+    {
+        key @0 :MetaProperty;
+        value @2 :Data;
+    }
+}
+
+struct Profile {
+    # homes, addresses, appId are moved to OwnProfile.data
```

Comment:
Homes, addresses and appId and probably revoked authorizations must be public and I think they should be present here somehow. `OwnProfile.data` should contain your encrypted private data which meets the wire only for backup, e.g. your contacts, app settings, personal notes, maybe event logs, etc.

Reply:
ok, got it. so the data is an encrypted version of the entire metadata for that profile?
does that mean that a profile is only instantiated upon request?
or do they pre-exist as wire objects in memory on the server?

Reply:
How we imagined this is that `Profile` contains everything that is public, while OwnProfile also contains data but only your private stuff and is encrypted. Applications might have a dedicated part in both to read and write but need to ask for permission e.g. to read private data not directly related to the app.